src/main/java/it/pagopa/pm/gateway/service/VposService.java

```diff
@@ -0,0 +1,260 @@
+package it.pagopa.pm.gateway.service;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import it.pagopa.pm.gateway.client.restapicd.RestapiCdClientImpl;
+import it.pagopa.pm.gateway.client.vpos.HttpClient;
+import it.pagopa.pm.gateway.client.vpos.HttpClientResponse;
+import it.pagopa.pm.gateway.dto.PatchRequest;
+import it.pagopa.pm.gateway.dto.creditcard.StepZeroRequest;
+import it.pagopa.pm.gateway.dto.creditcard.StepZeroResponse;
+import it.pagopa.pm.gateway.dto.vpos.AuthResponse;
+import it.pagopa.pm.gateway.dto.vpos.ThreeDS2Challenge;
+import it.pagopa.pm.gateway.dto.vpos.ThreeDS2Method;
+import it.pagopa.pm.gateway.dto.vpos.ThreeDS2Response;
+import it.pagopa.pm.gateway.entity.PaymentRequestEntity;
+import it.pagopa.pm.gateway.repository.PaymentRequestRepository;
+import it.pagopa.pm.gateway.utils.VPosRequestUtils;
+import it.pagopa.pm.gateway.utils.VPosResponseUtils;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.BooleanUtils;
+import org.apache.commons.lang3.ObjectUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.entity.ContentType;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.math.BigInteger;
+import java.util.*;
+
+import static it.pagopa.pm.gateway.constant.ApiPaths.REQUEST_PAYMENTS_CREDIT_CARD;
+import static it.pagopa.pm.gateway.constant.Messages.*;
+import static it.pagopa.pm.gateway.constant.VposConstant.*;
+import static it.pagopa.pm.gateway.dto.enums.PaymentRequestStatusEnum.*;
+import static it.pagopa.pm.gateway.dto.enums.TransactionStatusEnum.TX_AUTHORIZED_BY_PGS;
+import static it.pagopa.pm.gateway.dto.enums.TransactionStatusEnum.TX_REFUSED;
+import static it.pagopa.pm.gateway.utils.MdcUtils.setMdcFields;
+
+@Service
+@Slf4j
+public class VposService {
+    @Value("${vpos.response.urlredirect}")
+    private String responseUrlRedirect;
+
+    @Value("${vpos.requestUrl}")
+    private String vposUrl;
+
+    private static final List<String> VALID_CLIENT_ID = Arrays.asList("APP", "WEB");
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private static final String CAUSE = " cause: ";
+
+    @Autowired
+    private PaymentRequestRepository paymentRequestRepository;
+
+    @Autowired
+    private RestapiCdClientImpl restapiCdClient;
+
+    @Autowired
+    private VPosRequestUtils vPosRequestUtils;
+
+    @Autowired
+    private VPosResponseUtils vPosResponseUtils;
+
+    @Autowired
+    private HttpClient httpClient;
+
+    public ResponseEntity<StepZeroResponse> startCreditCardPayment(String clientId, String mdcFields, StepZeroRequest request) {
```

Comment:
Being a service and not a controller it should not be the service's responsibility to set the status code for the response